File: bulk_import.py
import urllib.request
import urllib.parse
import json
import os
import time

PEXELS_KEY = 'BNz0gB5lTLncIkKcuXkLEVanU5lQE731mMZEBxPUKUhmmPDvtjxYhO6H'
SUPABASE_URL = 'https://zgcqbvvvwbgpbgaofkmg.supabase.co'
SUPABASE_KEY = 'sb_secret_f8T3RAMmcuuAJ2F99z68-w_2UiamJYy'

if not PEXELS_KEY or not SUPABASE_URL or not SUPABASE_KEY:
    print("ERROR: Missing environment variables")
    exit(1)

CATEGORY_SLUG = "cars-motorcycles"
QUERIES = ["classic cars", "muscle cars", "motorcycles", "sports cars", "vintage cars"]
TARGET = 100
imported = 0

for query in QUERIES:
    if imported >= TARGET:
        break
    page = 1
    while imported < TARGET:
        url = "https://api.pexels.com/v1/search?query=" + urllib.parse.quote(query) + "&per_page=20&page=" + str(page)
        req = urllib.request.Request(url, headers={"Authorization": PEXELS_KEY})
        try:
            with urllib.request.urlopen(req) as r:
                data = json.loads(r.read())
        except Exception as e:
            print("Pexels error: " + str(e))
            break
        photos = data.get("photos", [])
        if not photos:
            break
        for photo in photos:
            if imported >= TARGET:
                break
            image_url = photo["src"]["large"]
            description = photo.get("alt", query)
            photographer = photo.get("photographer", "Unknown")
            prompt = description + " (pexels photo by " + photographer + ")"
            try:
                with urllib.request.urlopen(image_url) as r:
                    image_data = r.read()
                    content_type = r.headers.get("Content-Type", "image/jpeg")
            except Exception as e:
                print("Download error: " + str(e))
                continue
            ext = "png" if "png" in content_type else "jpg"
            filename = "stock/" + str(int(time.time()*1000)) + "-pexels." + ext
            upload_url = SUPABASE_URL + "/storage/v1/object/jpix-generated/" + filename
            upload_req = urllib.request.Request(upload_url, data=image_data, headers={"Authorization": "Bearer " + SUPABASE_KEY, "Content-Type": content_type}, method="POST")
            try:
                with urllib.request.urlopen(upload_req) as r:
                    pass
            except Exception as e:
                print("Upload error: " + str(e))
                continue
            public_url = SUPABASE_URL + "/storage/v1/object/public/jpix-generated/" + filename
            insert_url = SUPABASE_URL + "/rest/v1/generated_images"
            payload = json.dumps({"prompt": prompt, "image_url": public_url, "status": "pending_review", "category_slug": CATEGORY_SLUG}).encode()
            insert_req = urllib.request.Request(insert_url, data=payload, headers={"Authorization": "Bearer " + SUPABASE_KEY, "apikey": SUPABASE_KEY, "Content-Type": "application/json", "Prefer": "return=minimal"}, method="POST")
            try:
                with urllib.request.urlopen(insert_req) as r:
                    imported += 1
                    print("[" + str(imported) + "/" + str(TARGET) + "] " + description[:60])
            except Exception as e:
                print("Insert error: " + str(e))
                continue
            time.sleep(0.2)
        page += 1

print("Done! Imported " + str(imported) + " images into pending review queue.")

File: bulk-import.mjs
// Bulk import images from Unsplash into Supabase generated_images
// Fetches ~5 images per category, inserts as approved

const UNSPLASH_KEY = "hBipcZk-pxDSFrQS5bLRA-xeaHh7eg07lBnIu296V0Q";
const SUPABASE_URL = "https://zgcqbvvvwbgpbgaofkmg.supabase.co";
const SUPABASE_SERVICE_KEY = "sb_secret_f8T3RAMmcuuAJ2F99z68-w_2UiamJYy";

const CATEGORIES = [
  { slug: "animals",          query: "wild animals nature" },
  { slug: "anime",            query: "anime illustration art" },
  { slug: "beaches",          query: "tropical beach paradise" },
  { slug: "big-cats",         query: "lion tiger leopard big cat" },
  { slug: "bikinis",          query: "bikini beach summer" },
  { slug: "boxing-mma",       query: "boxing MMA martial arts fighter" },
  { slug: "cars-motorcycles", query: "classic cars motorcycles" },
  { slug: "celebrity",        query: "glamour fashion portrait" },
  { slug: "costume",          query: "costume cosplay creative" },
  { slug: "faith",            query: "faith church spiritual" },
  { slug: "fantasy",          query: "fantasy dragon warrior magic" },
  { slug: "female-models",    query: "female portrait fashion model" },
  { slug: "food",             query: "delicious food photography" },
  { slug: "funny",            query: "funny humor comedy" },
  { slug: "hip-hop",          query: "hip hop rap urban street" },
  { slug: "classic-cars",     query: "classic car" },
  { slug: "inspirational",    query: "inspirational sunrise mountains" },
  { slug: "lingerie",         query: "lingerie fashion boudoir" },
  { slug: "lowriders",        query: "lowrider custom car chicano" },
  { slug: "male-models",      query: "male portrait fitness model" },
  { slug: "military",         query: "military soldier army" },
  { slug: "miscellaneous",    query: "abstract art colorful" },
  { slug: "music",            query: "music concert performance" },
  { slug: "native-american",  query: "Native American culture heritage" },
  { slug: "nature",           query: "nature landscape scenic" },
  { slug: "old-school",       query: "vintage retro classic" },
  { slug: "pin-up",           query: "pin-up retro glamour" },
  { slug: "seasonal",         query: "seasons autumn winter spring" },
  { slug: "sports",           query: "sports athlete action" },
  { slug: "supercars",        query: "supercar ferrari lamborghini exotic car" },
  { slug: "tattoo-art",       query: "tattoo art flash design" },
  { slug: "western",          query: "cowboy western rodeo" },
  { slug: "wolves-eagles",    query: "wolf eagle power animal" },
  { slug: "yoga",             query: "yoga fitness wellness" },
];

const PER_CATEGORY = 55;
const PAGE = 2; // page 1 already imported — use 2 for fresh results

async function fetchUnsplash(query, perPage = 10, page = 2) {
  const url = `https://api.unsplash.com/search/photos?query=${encodeURIComponent(query)}&per_page=${perPage}&page=${page}&orientation=portrait&content_filter=high`;
  const res = await fetch(url, {
    headers: { Authorization: `Client-ID ${UNSPLASH_KEY}` },
  });
  if (!res.ok) {
    console.error(`Unsplash error for "${query}":`, res.status, await res.text());
    return [];
  }
  const data = await res.json();
  return data.results || [];
}

async function insertImages(images) {
  const res = await fetch(`${SUPABASE_URL}/rest/v1/generated_images?on_conflict=image_url`, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
      "apikey": SUPABASE_SERVICE_KEY,
      "Authorization": `Bearer ${SUPABASE_SERVICE_KEY}`,
      "Prefer": "return=representation,resolution=ignore-duplicates",
    },
    body: JSON.stringify(images),
  });
  if (!res.ok) {
    const err = await res.text();
    console.error("Supabase insert error:", err);
    return 0;
  }
  const data = await res.json();
  return data.length;
}

async function main() {
  let totalInserted = 0;

  for (const category of CATEGORIES) {
    console.log(`\nFetching "${category.query}" for [${category.slug}]...`);
    const photos = await fetchUnsplash(category.query, PER_CATEGORY, PAGE);

    if (photos.length === 0) {
      console.log("  No photos found, skipping.");
      continue;
    }

    const rows = photos.map((photo) => ({
      prompt: photo.description || photo.alt_description || category.query,
      image_url: photo.urls.regular,
      status: "pending_review",
      category_slug: category.slug,
    }));

    const count = await insertImages(rows);
    console.log(`  Inserted ${count} images.`);
    totalInserted += count;

    // Be polite to the API
    await new Promise((r) => setTimeout(r, 300));
  }

  console.log(`\nDone! Total images inserted: ${totalInserted}`);
}

main().catch(console.error);
